In [1]:
# !pip install feature_engine

In [2]:
# import lagi libraries yang dibutuhkan
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.api as sm
import pickle
import json
import time
import xgboost as xgb
from matplotlib import colors
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV
from sklearn.metrics import mean_squared_error, root_mean_squared_error, mean_absolute_error, r2_score
from sklearn.preprocessing import OneHotEncoder, StandardScaler, RobustScaler, MinMaxScaler
from sklearn.linear_model import LinearRegression
from sklearn.metrics import classification_report, f1_score, precision_score, accuracy_score, recall_score
from feature_engine.outliers import Winsorizer
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import make_pipeline
from sklearn.neighbors import KNeighborsRegressor
from sklearn.svm import SVR
from sklearn.ensemble import RandomForestRegressor, AdaBoostRegressor
from sklearn.tree import DecisionTreeRegressor

In [3]:
# panggil lagi model untuk inference
with open('list_num_cols.txt', 'rb') as file_1:
  num_col = json.load(file_1)

with open('list_cat_cols.txt', 'rb') as file_2:
  cat_col = json.load(file_2)

with open('num_pipeline.pkl', 'rb') as file_3:
  num_pipeline = pickle.load(file_3)

with open('cat_pipeline.pkl', 'rb') as file_4:
  cat_pipeline = pickle.load(file_4)

with open('preprocessing.pkl', 'rb') as file_5:
  preprocessing_pipeline = pickle.load(file_5)

with open('xgboost.pkl', 'rb') as file_6:
  xg = pickle.load(file_6)

with open('pipe_xg_1.pkl', 'rb') as file_7:
  pipe_xg = pickle.load(file_7)

In [4]:
# isi data baru untuk inference
x_inf = {
    'min_players':1,
    'max_players':4,
    'playtime':10,
    'min_age':10,
    'complexity':5,
    'owned_users':5,
    'top_mechanics':'Dice Rolling',
    'top_domains':"Children's Games"
}
x_inf_df = pd.DataFrame([x_inf])
x_inf_df

min_players  max_players  playtime  min_age  complexity  owned_users  \
0            1            4        10       10           5            5   

  top_mechanics       top_domains  
0  Dice Rolling  Children's Games

In [5]:
# lakukan prediksi rating pada data baru
pred_inf = pipe_xg.predict(x_inf_df)
pred_inf

array([6.7244377], dtype=float32)

Didapatkan rating 6.72 dari fitur-fitur game baru yang saya inputkan.